src/open_prime_hunters_rando/entities/pickup.py

```diff
@@ -1,70 +1,73 @@
+                        "message1_target": old_entity_data.notify_entity_id,
+                        "_padding1": 0,
+                        "message1": old_entity_data.collected_message,
+                        "message2_target": 0,
+                        "_padding2": 0,
+                        "message2": 0,
+                        "message3_target": 0,
+                        "_padding3": 0,
+                        "message3": 0,
+                        "linked_entity_id": -1,
+                    }
+                )
 
         # Update Artifact Entities
         # Entity was Artifact
         else:
             # Entity is still Artifact
-            if new_entity_type == old_entity_type:
-                entity_file[main_data] = pickup["model_id"]
-                entity_file[main_data + 1] = pickup["artifact_id"]
+            if new_entity_type == EntityType.ARTIFACT.value:
+                entity.data.model_id = pickup["model_id"]
+                entity.data.artifact_id = pickup["artifact_id"]
             # Entity is now ItemSpawn
             else:
-                # Moving similar fields to the new offsets
-                entity_file[main_data + 20] = entity_file[main_data + 8]  # message1
-                entity_file[main_data + 8] = entity_file[main_data + 2]  # active
-                entity_file[main_data + 9] = entity_file[main_data + 3]  # has_base
-                entity_file[main_data + 18] = entity_file[main_data + 4]  # message1_target
-
-                # Changes to match ItemSpawn entities
-                entity_file[main_data] = 0xFF  # Always FF
-                entity_file[main_data + 1] = 0xFF  # Always FF
-                entity_file[main_data + 2] = 0x00
-                entity_file[main_data + 3] = 0x00
-                entity_file[main_data + 4] = ITEM_TYPES_TO_IDS[pickup["item_type"]]
-                entity_file[main_data + 5] = 0x00
-                entity_file[main_data + 12] = 0x01  # max spawn count
-                entity_file[main_data + 13] = 0x00
-                entity_file[main_data + 19] = 0x00
-                entity_file[main_data + 21] = 0x00
-                entity_file[main_data + 28] = 0x00
-                entity_file[main_data + 29] = 0x00
+                entity.entity_type = EntityType.ITEM_SPAWN
+                entity.data = Container(
+                    {
+                        "header": header,
+                        "parent_id": 65535,
+                        "item_type": pickup["item_type"],
+                        "enabled": old_entity_data.active,
+                        "has_base": old_entity_data.has_base,
+                        "always_active": False,
+                        "_padding": 0,
+                        "max_spawn_count": 1,
+                        "spawn_interval": 0,
+                        "spawn_delay": 0,
+                        "notify_entity_id": old_entity_data.message1_target,
```

Comment:
add a TODO comment indicating that you need to fix instances where a vanilla artifact sends messages to more than 1 entity